src/com/puppetlabs/puppetdb/facts.clj

```diff
@@ -11,17 +11,37 @@
 
 ;; SCHEMA
 
+;; Terminology:
+;;   facts: {"kernel" "Linux" ...}
+;;   factmap: {:factset_id ... :fact_path_id ... fact_value_id ...}
+;;   fact-path path: ["foo" "bar"]
+;;   pathstr: "foo~#bar..."
+;;   pathmap: {... :path ... :name ... :depth ...}
+;;   valuemap: {... :value_type_id ... :value_boolean ...}
+
 (def fact-path-element
   (s/either s/Str s/Num))
 
 (def fact-path
   [fact-path-element])
 
-(def fact-path-map
+(def pathmap-scm
   {:path s/Str
-   :depth s/Int
    :name s/Str
-   :value_hash s/Str
+   :depth s/Int})
+
+(def fact-set
+  {s/Str s/Any})
+
+(def facts-schema
+  {:name String
+   :values fact-set
+   :timestamp pls/Timestamp
+   :environment (s/maybe s/Str)
+   :producer-timestamp (s/either (s/maybe s/Str) pls/Timestamp)})
```

Comment:
@wkalt, @senior well that might be more luck than anything else.  If you mean the var names too, I hadn't realized/remembered that we'd carried it that far.  i.e. I'd thought we'd kept dashes for function args, and other "public", top-level bits.  In any case, glad you pointed it out.  I (or whoever does it) will need to keep that in mind during the merge to master.